### PR TITLE
Add Objects365 Dataset with Optimized Download Process

### PR DESCRIPTION
**Pull request by hglee98**
## Description
This PR introduces support for the Objects365 dataset, a large-scale dataset crucial for enhancing our promising RT-DETR model's object detection performance. Given the dataset's substantial size, I've implemented several optimizations to streamline the download process.

### Key Improvements

- **Multi-processing Downloads**: Implemented a multi-processing feature to reduce overall download time for the numerous data patches.
- **Robust Connection Handling**: Added max-timeout and connect-timeout options to manage intermittent connection issues between host and client, ensuring download continuity.
- **New Download Script**: Created `tools/open_dataset_tool/objects365.py` for efficient dataset acquisition.

Closes: #481

## Change(s)

- Add download script for the objects365 `tools/open_dataset_tool/objects365.py`.
- Add multi-processing option when downloading the dataset since there are too many data patches to download.
- Add max-timeout and connect-timeout options at the downloading line because the connection between host and client is disconnected and downloading progress is stopped sometimes.
- Update corresponding docs
- Add predefined-configuration

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`. 

## Changelog

If your PR is granted to `dev` branch, codeowner will add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file including a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

Please enable **Allow edits and access to secrets by maintainers** so that our maintainers can update the `CHANGELOG.md`.

### CODE REVIEW

File: tools/open_dataset_tool/objects365.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                "--max-time", "300",
+                "-C", "-",
+                url
+            ], check=True)
+            print(f"Successfully downloaded {url} to {path}")
+            return
+        except subprocess.CalledProcessError as e:
+            print(f"Attempt {attempt + 1} failed to download {url}: {e}")
+            if attempt < max_retries - 1:
+                time.sleep(delay)
```
Comment (illian01) [inline tools/open_dataset_tool/objects365.py:98]: Can you add retry log more explicitly? For example, 

```suggestion
        except subprocess.CalledProcessError as e:
            print(f"Attempt {attempt + 1} failed to download {url}: {e}")
            print(f"Retry download after {delay} seconds ...")
            if attempt < max_retries - 1:
                time.sleep(delay)
```

File: tools/open_dataset_tool/objects365.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import torch
+import shutil
+import argparse
+import subprocess
+from pathlib import Path
+from tqdm import tqdm
+from multiprocessing import Pool
+
+DEFAULT_DATA_DIR = './data'
+DOWNLOAD_DIR = './data/download'
```
Comment (illian01) [inline tools/open_dataset_tool/objects365.py:29]: I think Objects365 dataset has so many data split to download. How about changing download target path to `./data/download/objects365`?

```suggestion
DOWNLOAD_DIR = './data/download/objects365'
```

**Review by illian01 — CHANGES_REQUESTED** [review]
I have few suggests about this PR.
Please check detailed review.

**Comment by hglee98** [comment]
Thank you for your kind review! I appreciate it :) I reflect your review and revise the source codes.

**Review by illian01 — APPROVED** [review]
LGTM!

**Pull request merged**